### ISSUE TITLE
[Feature request]: Give a hint when an app installed system-wide depends on a runtime only found per-user

### ISSUE THREAD

**Issue by K0-RR** [flatpak/flatpak#4730]
### Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/flatpak/flatpak/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/flatpak/flatpak/issues) for a feature request that matches the one I want to file, without success.

### Suggestion

Package installed as per-user is not suitable for an application that is system-wide if I understand correctly. The error message doesn't specify this and doesn't say how to fix this, instead it states that the runtime was not found.

Originated in https://github.com/flatpak/flatpak/issues/4724#issuecomment-1030512949. Please excuse me that I didn't explain it here but I don't know how to make it sound technical/proper.

**Comment by smcv** [comment]
I don't think that error message can really be improved **in general**, because all we know is "you need runtime x/y/z, but you don't have runtime x/y/z, so that can't work".

Am I understanding correctly that you are asking for a nicer error message **in one specific situation**: when you are trying to install an application system-wide, the required runtime cannot be found system-wide, but you do have that runtime available in your own personal Flatpak installation location `~/.local/share/flatpak`?

Something like this might be achievable, for instance:

```
error: The application com.brave.Browser/x86_64/beta requires the runtime org.freedesktop.Platform/x86_64/21.08 which was not found
hint: Required runtime org.freedesktop.Platform/x86_64/21.08 is available per-user but not system-wide
hint: To install the application system-wide, try adding the remote "flathub" system-wide
hint: Or to install the application per-user, try: flatpak install --user com.brave.Browser/x86_64/beta
```

Unfortunately I'm not sure that there's enough information stored in remotes to be able to show a sample command-line to add it system-wide, without being a really intimidating command-line (it would have lots of options like `--title`, `--icon`, `--homepage`, `--gpg-import` which we probably don't want to spam you with here).
